pkg/pf/tfbridge/provider_upgrade_resource_state.go

```diff
@@ -28,21 +31,27 @@ import (
 func (p *provider) UpgradeResourceState(
 	ctx context.Context,
 	rh *resourceHandle,
-	st *resourceState,
+	st *rawResourceState,
 ) (*upgradedResourceState, error) {
-	if st.TFSchemaVersion >= rh.schema.ResourceSchemaVersion() {
```

Comment:
One change here is that this always now run the upgrade method even when versions indicate that the resource state being upgraded is on a current enough schema. I wonder whether this is wrong though in terms of parity with Terraform. Let me check if Terraform runs the upgrade method always or not.

Reply:
https://github.com/opentofu/opentofu/blob/4083fe2cc4a483fc19d9769e0eb19ebee2386b33/internal/tofu/upgrade_resource_state.go#L28  seems like this is in fact conditional.. so the check needs to be brought back, and worked around/tested.

Reply:
Actually reading the code carefully today, TF does always call the upgrade method even if current schema numbers match. This is great. This makes implementation easier for us I just need to annotate.

Reply:
We might need to have it under flag though.